### ISSUE TITLE
Ctrl+Space shows documentation tooltip for wrong function

### ISSUE THREAD

**Issue by hickford** [ionide/ionide-vscode-fsharp#806]
Hi. Can anyone else reproduce this bug? With the code below, try pressing Ctrl+Space after `Seq.map`. You will see a tooltip with documentation for the Seq.map function. Then, try pressing Ctrl+Space after `List.map`. Expected: You will see a tooltip with documentation for List.map. What actually happens: You see a tooltip with documentation for Seq.map , ie. the wrong function!

    let x = Seq.map
    let y = List.map

If you just hover on the functions, you see the right documentation

![image](https://user-images.githubusercontent.com/105314/39957784-5868558c-55f0-11e8-99b1-1e95b9d924ce.png)

**Comment by inosik** [comment]
What's even stranger, this happens with the `Option` module as well:

![Seq.map instead of Option.map](https://user-images.githubusercontent.com/11393003/39984569-f10e79b2-575b-11e8-8ffb-7fa742d4c44a.png)

**Comment by inosik** [comment]
To reproduce this:

1. Add these modules:
    ``` fsharp
    module M =
      let f = ()

    module M2 =
      let f x y = ()
    ```

1. Access `M.f` in the editor

1. Access `M2.f` in the editor

    _Tooltip for `M.f` is shown, instead of `M2.f`._

Note that only <kbd>Ctrl</kbd>+<kbd>Space</kbd> is affected, hovering over a member shows the correct tooltip.

**Comment by inosik** [comment]
The request on <kbd>Ctrl</kbd>+<kbd>Space</kbd> for `Seq.collect` looks like this:

``` text
[10:22:19 DEBUG] REQ (15132) -> {helptext}, Symbol = "collect"
Data={"Symbol":"collect"}
[10:22:19 DEBUG] RES (15132) <- {helptext} in 4 ms: Kind={"helptext"}
Data={"Name":"collect","Overloads":[[{"Signature":"val collect : mapping:(char -> string) -> str:string -> string","Comment":""}]],"AdditionalEdit":null}
``` 

1. `Symbol = "collect"` is not enough information to get the correct function
2. FSAC caches the result and the cache isn't invalidated again

    _Whichever `collect` gets cached first, is returned for every other `collect`._

**Comment by Krzysztof-Cieslak** [comment]
It's something wrong with FSAC tooltips caching, I belive

**Comment by Krzysztof-Cieslak** [comment]
Fixed in `3.20.7`

**Comment by hickford** [comment]
Brilliant, thanks very much!
